test/functional/tests/startup/test_startup_switch_kernels.py

```diff
@@ -0,0 +1,257 @@
+            TestRun.fail("Core is not mounted!")
+
+    with TestRun.step("Check if md5 checksum matches."):
+        md5_after = test_file.md5sum()
+        if md5_before != md5_after:
+            TestRun.fail("Md5 checksum mismatch!")
+
+    with TestRun.step("Change back kernel subversion."):
+        switch_kernel(custom=previous_kernel)
+
+    with TestRun.step("Reboot"):
+        TestRun.executor.reboot()
+
+    with TestRun.step("Test cleanup."):
+        fstab.remove_mountpoint(device=core)
+        core.unmount()
+        InitConfig.create_default_init_config()
+        casadm.stop_all_caches()
+
+
+@pytest.mark.remote_only
+@pytest.mark.require_incompatible_kernel
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_cas_startup_incompatible_kernel():
+    """
+    title: Test for starting CAS on system startup after reboot to compatible kernel.
+    pass_criteria:
+      - System does not crash.
+      - CAS modules are loaded before partitions are mounted.
+      - Cache is loaded before partitions are mounted.
+      - Exported object is mounted after startup is complete.
+    """
+    with TestRun.step("Prepare partitions for cache and for core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(200, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(400, Unit.MebiByte)])
+        core_part = core_dev.partitions[0]
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, force=True)
+
+    with TestRun.step("Create filesystem, add core and mount it."):
+        core_part.create_filesystem(Filesystem.ext4)
+        core = cache.add_core(core_part)
+        core.mount(mountpoint)
+
+    with TestRun.step("Create test file and calculate md5 checksum."):
+        test_file = fs_utils.create_random_test_file(filepath, Size(16, Unit.MebiByte))
+        md5_before = test_file.md5sum()
+
+    with TestRun.step("Add mountpoint fstab and create opencas.conf."):
+        fstab.add_mountpoint(device=core,
+                             mount_point=mountpoint,
+                             fs_type=Filesystem.ext4)
+        InitConfig.create_init_config_from_running_configuration()
+
+    with TestRun.step("Change kernel subversion in boot configuration."):
+        previous_kernel = check_current_kernel()
+        switch_kernel(compatible=False)
+
+    with TestRun.step("Reboot."):
+        TestRun.executor.reboot()
+
+    with TestRun.step("Check if cache is started."):
+        caches = list(get_caches())
+        if len(caches) != 1:
+            pytest.xfail(f"Should not can list caches.")
+
+    with TestRun.step("Check if core is added."):
+        cores = list(get_cores(cache.cache_id))
+        if len(cores) != 1:
+            pytest.xfail(f"Should not can list cores.")
```

Comment:
"should not can" - it is not a correct syntax. Please fix in in the whole patch.